sirepo/package_data/static/js/activait.js

```diff
@@ -1096,20 +1096,44 @@ SIREPO.app.directive('imagePreviewPanel', function(requestSender) {
         restrict: 'A',
         scope: {
             method: '@',
-            imageClass: '@',
         },
         template: `
-        <div>
-          <img class="img-responsive {{ imageClass }}" />
+        <div class="container-fluid">
+          <div data-ng-if="x" class="row">
+            <div class="col-md-{{ hasThirdColumn ? '4' : '6' }}">
+              <div class="lead text-center">{{ xColName }}</div>
+            </div>
+            <div class="col-md-{{ hasThirdColumn ? '4' : '6' }}">
+              <div class="lead text-center">{{ yColName }}</div>
+            </div>
+            <div data-ng-if="hasThirdColumn" class="col-md-4">
+              <div class="lead text-center">{{ predColName }}</div>
+            </div>
+          </div>
+          <div data-ng-repeat="image in pageImages">
+            <div class="row">
+              <div class="col-md-{{ hasThirdColumn ? '4' : '6' }}">
+                <img class="img-responsive x{{ method + ($index + 1) }}" />
+                <div data-ng-if="xIsParams"> <br/> <b>{{ parameters[$index] }}</b> </div>
+              </div>
+              <div class="col-md-{{ hasThirdColumn ? '4' : '6' }}">
+                <img class="img-responsive y{{ method + ($index + 1) }}" />
+                <div data-ng-if="imageToLabels" class="text-center"> <br/> <b>{{ labels[$index] }}</b> </div>
+              </div>
+              <div data-ng-if="hasThirdColumn" class="col-md-4">
+                <img class="img-responsive pred{{ method + ($index + 1) }}" />
+              </div>
+            </div>
+          </div>
           <div data-ng-if="isLoading()" data-sim-state-progress-bar="" data-sim-state="simState"></div>
           <div data-ng-if="dataFileMissing">Data file {{ fileName }} is missing</div>
           <div data-ng-if="! isLoading() && multiPage">
-            <div class="pull-left">
+            <div data-ng-if="numPages > 3" class="pull-left">
               <button class="btn btn-primary" title="first" data-ng-click="first()">|<</button>
               <button class="btn btn-primary" title="previous" data-ng-disabled="! canUpdateUri(-1)" data-ng-click="prev()"><</button>
             </div>
-            <div class="pull-right">
-                page {{ idx + 1 }} of {{ uris.length }}
+            <div data-ng-if="numPages > 3" class="pull-right">
+                page {{ imageIdx / imagesPerPage + 1 }} of {{ numPages }}
```

Comment:
This page counting is not robust against changes in total number or images or images per page. e.g. change `$scope.imagesPerPage` to 4 ("Page 1 of 3.75"), click to go to the end ("Page 3.75 of 3.75") and then start decrementing - at "Page 0.75 of 3.75" you get a 404 error and a broken image.